### PR TITLE
feat(store): add ngrx-store-freeze migration

### LINKED ISSUE #1896: Store: add migration to move users off ngrx-store-freeze
The majority of our users are probably using `ngrx-store-freeze` to guard against state mutations. We have built-in run-time checks in version 8 to guard against those. This migration would

- Check to see if they are using `ngrx-store-freeze`
  - Remove `ngrx-store-freeze` from imports and the meta-reducers array
  - Remove `ngrx-store-freeze` from their `package.json` dependencies.
- Update StoreModule.forRoot() to enable the `strictStateImmutability` and `strictActionImmutability` run-time checks.

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #1896 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

Still to be done:

- [x] Add it to the v8 migration
- [x] Cleanup

### CODE REVIEW

File: modules/store/migrations/migration.json [hunk 1]
```diff
@@ -8,9 +8,14 @@
...
     "ngrx-store-migration-02": {
-      "description": "The road to v8",
+      "description": "The road to v8 beta",
       "version": "8-beta",
-      "factory": "./8_0_0/index"
+      "factory": "./8_0_0-beta/index"
+    },
+    "ngrx-store-migration-03": {
+      "description": "The road to v8 RC",
+      "version": "8-rc.1",
```
Comment (timdeschryver) [inline modules/store/migrations/migration.json]: Will this be picked up? 🤔

File: modules/data/schematics-core/utility/ast-utils.ts [hunk 2]
```diff
@@ -636,3 +643,52 @@ export function insertImport(
...
+      if (
+        specifier.propertyName &&
+        specifier.propertyName.text === importAsIs
+      ) {
+        return { hit: true, specifier, text: specifier.propertyName.text };
+      }
+
+      return { hit: false };
+    })
+    .filter(({ hit }) => hit)
```
Comment (ThomasBurleson) [inline modules/data/schematics-core/utility/ast-utils.ts]: Nice.

File: modules/effects/schematics-core/utility/ast-utils.ts [hunk 3]
```diff
@@ -636,3 +643,52 @@ export function insertImport(
...
+
+  if (imports.length === 0) {
+    return [];
+  }
+
+  const changes = imports
+    .map(p => (p.importClause!.namedBindings! as ts.NamedImports).elements)
+    .reduce((imports, curr) => imports.concat(curr), [] as ts.ImportSpecifier[])
+    .map(specifier => {
+      if (!ts.isImportSpecifier(specifier)) {
```
Comment (ThomasBurleson) [inline modules/effects/schematics-core/utility/ast-utils.ts]: Is this not a duplicate of code in `ast-utils.ts`#L666

File: modules/store-devtools/schematics-core/utility/ast-utils.ts [hunk 4]
```diff
@@ -636,3 +643,52 @@ export function insertImport(
...
+      ({ moduleSpecifier }) =>
+        moduleSpecifier.getText(sourceFile) === `'${importFrom}'` ||
+        moduleSpecifier.getText(sourceFile) === `"${importFrom}"`
+    );
+
+  if (imports.length === 0) {
+    return [];
+  }
+
+  const changes = imports
```
Comment (ThomasBurleson) [inline modules/store-devtools/schematics-core/utility/ast-utils.ts]: Seems totally NOT DRY; or am I mistaken?

File: modules/effects/schematics-core/utility/ast-utils.ts (hunk 3 above)
Comment (timdeschryver) [reply modules/effects/schematics-core/utility/ast-utils.ts]: Do you mean that this code is repeated in all of the modules? 
This is by design. If I'm not mistaken by doing it this way, we don't need to create a schematics-core package. (see #1025 and #1044).

File: modules/store/migrations/migration.json (hunk 1 above)
Comment (brandonroberts) [reply modules/store/migrations/migration.json]: 🤷‍♂

Comment (brandonroberts) [reply modules/store/migrations/migration.json]: This should work according to: https://github.com/angular/angular-cli/blob/master/packages/schematics/update/migrate/index_spec.ts#L133

**Pull request merged**